lib/assessment.js

```diff
@@ -40,8 +58,12 @@ var newArray2 = function newArray2(length, defaultsFunction) {
 // the object passed in as an argument or the object created
 var addMethod = function addMethod(propertyName, method, obj) {
 
+//i have no idea how to do this on my own
```

Comment:
No worries!
Here's the idea, step by step.
1. Check to see if `obj` has been defined; set it to a blank object to start out with.
2. Add a new method to that `obj` with the name given.
3. Return the modified object.

Here's how that might look in code.
```javascript
if (!obj) { // if obj is undefined
  obj = {};
}
obj[propertyName] = method;
return obj;
```

That first bit could also have been written as `obj = obj || {}`, since `||` will evaluate to the first truthy argument it gets.